psc-ide.el

```diff
@@ -229,13 +232,20 @@ COMMAND, ARG and IGNORED correspond to the standard company backend API."
        (unless (or
                 ;; Don't add an import when the option to do so is disabled
                 (not psc-ide-add-import-on-completion)
-                ;; or when a qualified identifier was completed
-                (or (get-text-property 0 :qualifier arg) (s-contains-p "." (company-grab-symbol)))
                 ;; Don't attempt to import Prim members
                 (string= (get-text-property 0 :module arg) "Prim"))
-         (psc-ide-add-import-impl arg (vector
-                                       (psc-ide-filter-modules
-                                        (list (get-text-property 0 :module arg))))))))))
+         (if (not (null unqualified-import-completion-qualifier))
```

Comment:
This is all in conflict now, sorry, because the containing code's global modification of `after-save-hook` was totally broken, and I've fixed that separately for the existing code.